AgileBoard.Api/src/AgileBoard.Api/Controllers/V1/UserStoriesController.cs

```diff
@@ -0,0 +1,115 @@
+﻿using AgileBoard.Api.Contracts.Requests;
+using AgileBoard.Api.Contracts.Responses;
+using AgileBoard.Api.Contracts.V1;
+using AgileBoard.Api.Domain;
+using AgileBoard.Api.Extensions;
+using AgileBoard.Api.Services;
+using AutoMapper;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.Mvc;
+
+namespace AgileBoard.Api.Controllers.V1
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class UserStoriesController : ControllerBase
+    {
+        private readonly IUserStoryService _userStoryService;
+        private readonly IMapper _mapper;
+
+        public UserStoriesController(IUserStoryService userStoryService, IMapper mapper)
+        {
+            _userStoryService = userStoryService;
+            _mapper = mapper;
+        }
+
+        [HttpGet(ApiRoutes.UserStories.GetAll)]
+        public async Task<IActionResult> GetAllAsync()
+        {
+            var userStories = await _userStoryService.GetUserStoriesAsync();
+            var result = _mapper.Map<IEnumerable<UserStoryResponse>>(userStories);
+            return Ok(result);
+        }
+
+        [HttpGet(ApiRoutes.UserStories.Get)]
+        public async Task<IActionResult> GetAsync([FromRoute] Guid userStoryId)
+        {
+            var userStory = await _userStoryService.GetUserStoryByIdAsync(userStoryId);
+            if(userStory == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(_mapper.Map<UserStoryResponse>(userStory));
+        }
+
+        [HttpPost(ApiRoutes.UserStories.Create)]
+        public async Task<IActionResult> CreateAsync([FromBody] CreateUserStoryRequest request)
+        {
+            var userStory = new UserStory
+            {
+                Id = Guid.NewGuid(),
+                Title = request.Title,
+                Owner = request.Owner,
+                Description = request.Description,
+                AcceptanceCriteria = request.AcceptanceCriteria,
+                DefinitionOfDone = request.DefinitionOfDone,
+                DefintionOfReady = request.DefintionOfReady,
+                Comments = request.Comments,
+                StoryPoints = request.StoryPoints,
+                Priority = request.Priority,
+                Risk = request.Risk,
+                Deadline = request.Deadline,
+                Created = DateTime.UtcNow,
```

Comment:
do not use DateTime.UtcNow create IClock or something that will give you abstraction over DateTime, if you will beed to test it, it's way easier to mock abstraction

Reply:
added ICock